flexmeasures/cli/tests/conftest.py

```diff
@@ -100,32 +103,50 @@ def reporter_config_raw(app, db, setup_dummy_data):
 
 @pytest.mark.skip_github
 @pytest.fixture(scope="module")
-def process_power_sensor(db, app):
+def process_power_sensor(db, app, setup_markets, setup_sources):
     """
-    Create an asset of type "ProcessType" and a power sensor to hold the result of
-    the scheduler.
+    Create an asset of type "process", power sensor to hold the result of
+    the scheduler and price data consisting of 8 expensive hours, 8 cheap hours, and again 8 expensive hours-
 
     """
 
     process_asset_type = GenericAssetType(name="process")
 
     db.session.add(process_asset_type)
 
-    processasset = GenericAsset(
-        name="Test Asset", generic_asset_type=process_asset_type
+    process_asset = GenericAsset(
+        name="Test Process Asset", generic_asset_type=process_asset_type
     )
 
-    db.session.add(processasset)
+    db.session.add(process_asset)
 
     power_sensor = Sensor(
         "power",
-        generic_asset=processasset,
+        generic_asset=process_asset,
         event_resolution=timedelta(hours=1),
         unit="MW",
     )
 
     db.session.add(power_sensor)
 
+    time_slots = initialize_index(
+        start=pd.Timestamp("2015-01-02").tz_localize("Europe/Amsterdam"),
+        end=pd.Timestamp("2015-01-03").tz_localize("Europe/Amsterdam"),
+        resolution="1H",
+    )
+    values = [100] * 8 + [90] * 8 + [100] * 8
+    beliefs = [
+        TimedBelief(
+            event_start=dt,
+            belief_horizon=timedelta(hours=0),
+            event_value=val,
+            source=setup_sources["Seita"],
+            sensor=setup_markets["epex_da"].corresponding_sensor,
```

Comment:
[Does this sensor not already have these price data for 2nd of January 2015](https://github.com/FlexMeasures/flexmeasures/blob/main/flexmeasures/conftest.py#L537) and you should n't we use that fixture?

Reply:
Good point!

In a previous iteration, the test code the fixture that you mentioned. It run find when running the test alone, but failed when running the whole suite (the magic Instance not in Session error). Using trial and error, I found that `test_add_annotation` and `test_add_holidays` use the fixture `setup_roles_users` which creates the users and roles. This same fixture is used in `add_market_prices` through the fixture  `setup_assets`. That's why I just added some lines of code to add the price data.

Reply:
I don't understand the problem yet.
Is it that these other tests (using the fixtures you mentioned) somewhere delete this data?
Then a nicer solution would be to use the fresh database fixtures (we usually make pairs, one which has everything made fresh, others that save time by not doing that and using whatever the last test left)

Reply:
Thanks for pointing out to the fresh_db fixtures. Last week, I tried with adding them to the new tests but I couldn't make it work.

I've just pushed a commit that changes to use a fresh_db in the tests `test_add_annotation` and `test_add_holidays`. With these changes, I removed the piece of code that added the timely beliefs :smile:

Reply:
Alright, better. But I still don't see what the problem was. (and why we need to add running time for db creation to these other tests).

test_add_annotation and test_add_holidays don't seem to mess with price data, so that is not it.

I played around and got the dreaded `sqlalchemy.orm.exc.DetachedInstanceError`. Is that the one you also saw?

Reply:
Yes... that's the one :S 

The issue is that the two tests are using `setup_roles_users` fixture which setup the users and the `add_market_prices` use that same fixture. When I call that fixture, the User object has already been fetched by the previous tests but gets detached from the session.

Reply:
Can we think of a fix similar to the fixes we did earlier for the same kind of error?

Reply:
There's one "dirty" fix consist of returning the id instead of the object. That would avoid the detached instance error.

Reply:
I had some success with refreshing the user object in our simulate package, not sure if that would help you here.

```
db.session.refresh(user)
```

Reply:
Thanks for helping me!

I tried that but it didn't succeed. A workaround that I tried is to return the sensor id's instead of the sensor instances and then use `Sensor.query.get(sensor_id)' to get the object back. I tried to see if the tests run in CI removing the skip decorators, but I saw an error similar to https://github.com/FlexMeasures/flexmeasures/pull/778.

Please, let me know if the changes I made are good to close this conversion :smile: